### ISSUE TITLE
CI: Non-deterministic failure of curl in macOS brew installation (py38)

### ISSUE THREAD

**Issue by RickyC0626** [ManimCommunity/manim#1303]
#### Preliminaries

- [X] I have followed the latest version of the [installation instructions](https://docs.manim.community/en/stable/installation.html).
- [X] I have checked the [troubleshooting page](https://docs.manim.community/en/stable/installation/troubleshooting.html) and my problem is either not mentioned there, or the solution given there does not help.

## Description of error

In specifically Python 3.8 macOS CI builds, `curl` experiences connection errors during the dependency installation step at seemingly random intervals.

## Installation logs

<details><summary>Terminal output</summary>

https://github.com/ManimCommunity/manim/runs/2305968662
```
==> Downloading https://homebrew.bintray.com/bottles/libbluray-1.3.0.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/acfe4417abf3c5169e7271e81043fd4fb6d26bcd2ea96a266b820c33e8492ac6?response-content-disposition=attachment%3Bfilename%3D%22libbluray-1.3.0.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvYWNmZTQ0MTdhYmYzYzUxNjllNzI3MWU4MTA0M2ZkNGZiNmQyNmJjZDJlYTk2YTI2NmI4MjBjMzNlODQ5MmFjNj9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMmxpYmJsdXJheS0xLjMuMC5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjE3OTczNTcyfSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=gn011oUzPV0IDn9Ee2fjgp20S~b5W8b~6wjy7QFNUA6Oj29Pp4up~HUIsSS-LJlooffKHTxV8cw7NHr7tgXVM6Z~17RwtiEri1niG-CVc0a0d5ewRruAF8tPgbdZ1k0Id~wJJOzLMHlLjvjBNYo8J~DiEhdGzFnW6Ag7b62sJ~P~bzXPz5g881iK4zGoa~sxcEUYZi6IjTuTR0TMaS-jtXrUER9sXvsK90zsCcXqk369lYjrumV-ry8gBKU02P3-nRdTkok~UPgyivuZCeloNzPjh3677zPNd3p4yXiAfUSurlv81FfMeamUmkrT7g1Chg9~FGf2XIDCdWpE8ueYXQ__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
curl: (35) LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to d29vzk4ow07wi7.cloudfront.net:443 
Error: Failed to download resource "libbluray"
Download failed: https://homebrew.bintray.com/bottles/libbluray-1.3.0.catalina.bottle.tar.gz
Error: Process completed with exit code 1.
```

https://github.com/ManimCommunity/manim/runs/2309595652
```
==> Downloading https://homebrew.bintray.com/bottles/pyenv-1.2.26.catalina.bottle.tar.gz
==> Downloading from https://d29vzk4ow07wi7.cloudfront.net/6a5736817f87bfbf97f355975a71ce99c72cff0afb8f5e29920c8eaac003f0ca?response-content-disposition=attachment%3Bfilename%3D%22pyenv-1.2.26.catalina.bottle.tar.gz%22&Policy=eyJTdGF0ZW1lbnQiOiBbeyJSZXNvdXJjZSI6Imh0dHAqOi8vZDI5dnprNG93MDd3aTcuY2xvdWRmcm9udC5uZXQvNmE1NzM2ODE3Zjg3YmZiZjk3ZjM1NTk3NWE3MWNlOTljNzJjZmYwYWZiOGY1ZTI5OTIwYzhlYWFjMDAzZjBjYT9yZXNwb25zZS1jb250ZW50LWRpc3Bvc2l0aW9uPWF0dGFjaG1lbnQlM0JmaWxlbmFtZSUzRCUyMnB5ZW52LTEuMi4yNi5jYXRhbGluYS5ib3R0bGUudGFyLmd6JTIyIiwiQ29uZGl0aW9uIjp7IkRhdGVMZXNzVGhhbiI6eyJBV1M6RXBvY2hUaW1lIjoxNjE4MDA3MDk1fSwiSXBBZGRyZXNzIjp7IkFXUzpTb3VyY2VJcCI6IjAuMC4wLjAvMCJ9fX1dfQ__&Signature=YyTUPkZa4kf7RirY43WSkVal2r03yhTbM6ihjIi4YA~ChVzrraqUlc2CweaQbro3XiVwh8QxEBFnDjzl~tT4O3Gg7im8Svf0pnwVc-sUN0V1QhZ~rgZEQRtJCDYu9lvny2UqHGUsP-yGtF7s3~MV53GeJmi0Ise4B0l2axKg4XMuhI5izAy1oQwpfq4yZhLyy7U59sBO3fEuQwuvcsNx1Og33gwW2MSgMhxMo~NJewzL075WhmCs88DcQwsROcm3LFMwiZ6ZSd0rEWuLOiDPFI6RbNwxpNdkexDindK~NtSQk7DkTWj8qoMfjJzYxdK2NwZvOFBlDBmuAwLS~mZ71w__&Key-Pair-Id=APKAIFKFWOMXM2UMTSFA
==> Pouring pyenv-1.2.26.catalina.bottle.tar.gz
🍺  /usr/local/Cellar/pyenv/1.2.26: 747 files, 2.6MB
==> Downloading https://github.com/pyenv/pyenv-virtualenv/archive/v1.1.5.tar.gz
==> Downloading from https://codeload.github.com/pyenv/pyenv-virtualenv/tar.gz/v1.1.5
curl: (7) Failed to connect to codeload.github.com port 443: Operation timed out
Error: Failed to download resource "pyenv-virtualenv"
Download failed: https://github.com/pyenv/pyenv-virtualenv/archive/v1.1.5.tar.gz
Error: Process completed with exit code 1.
```

</details>

## System specifications

<details><summary>System Details</summary>

According to [About GitHub-hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners):
- OS: macOS Catalina 10.15 (macos-latest)
- RAM: 14 GB
- Python version: 3.8
- Installed modules: [Virtual environment preinstalled software](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md)
   - Homebrew 3.0.10
   - Curl 7.76.0
</details>

## Additional comments

It may not be a problem related to Manim, but this provides documentation in case it happens a third time.

**Comment by jsonvillanueva** [comment]
@RickyC0626 Can this be closed now that #1257 is merged?

**Comment by RickyC0626** [comment]
> @RickyC0626 Can this be closed now that #1257 is merged?

Yeah, I think this problem shouldn't occur anymore.
